proto/anki/themes.proto

```diff
@@ -0,0 +1,148 @@
+// Copyright: Ankitects Pty Ltd and contributors
+// License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+
+syntax = "proto3";
+
+option java_multiple_files = true;
+
+package anki.themes;
+
+import "anki/generic.proto";
+import "anki/collection.proto";
+import "anki/decks.proto";
+
+service ThemesService {
+  rpc GetTheme(ThemeId) returns (Theme);
+  rpc RemoveTheme(ThemeId) returns (generic.Empty);
+  rpc GetThemesForUpdate(generic.Empty) returns (ThemesForUpdate);
+  rpc UpdateThemes(UpdateThemesRequest) returns (collection.OpChanges);
+}
+
+message ThemeId {
+  int64 tid = 1;
+}
+
+message Vars {
+  Factors factors = 1;
+  Colors colors = 2;
+}
+
+message Theme {
+  int64 id = 1;
+  string name = 2;
+  int64 mtime_secs = 3;
+  int32 usn = 4;
+  Vars vars = 5;
+}
+
+message Factors {
+  float font_size = 1;
+  float roundness = 2;
+  float spacing = 3;
+  float transparency = 4;
+  float animation_speed = 5;
+
+  bytes other = 255;
+}
+
+message Color {
+  float red = 1;
+  float green = 2;
+  float blue = 3;
+  float alpha = 4;
+}
+
+
+/* User-modifiable base colors
+from which helper colors (gradient, hover, etc.) are derived */
+message Colors {
+  // background
+  Color canvas = 1;
+  Color canvas_secondary = 2;
+  Color canvas_code = 3;
+  reserved 4;
+  reserved 5;
```

Comment:
I recommend cutting out these reserved lines - it doesn't matter if the numbers are out of order when we add new fields in the future. The main reason to reserve numbers is when we anticipate repeated fields to be required, as fields in 1-15 take 1 byte per entry, and 16+ take 2 bytes per entry, so for large lists it's more space efficient to place fields at the start.